en/docs/cookies/serving.md

```diff
@@ -0,0 +1,267 @@
+
+![](./imgs/oneflow-serving-demo.png)
+
+Analyzing the code, we can find the following key points:
+
+- Triton server and WEB application server are started in `run_cloud.sh`:
+```bash
+/opt/tritonserver/bin/tritonserver --model-store $(pwd)/model_repo > 1.txt &
+  python3 server.py
+```
+
+- `server.py` is just simple and normal URL routing, so `stylize` in `infer.py` does the real inference work. Its result is obtained inside the `stylize` function through interacting with the HTTP and Triton server.
+```python
+def stylize(content_path, output_path, style='udnie'):
+    triton_client = httpclient.InferenceServerClient(url='127.0.0.1:8000')
+    ...
+    inputs.append(httpclient.InferInput('INPUT_0', image.shape, 'FP32'))
+    ...
+    outputs.append(httpclient.InferRequestedOutput('OUTPUT_0', binary_data=True))
+    ...
+```
+
+- Pretrained models are placed under `model_repo`, which is organized according to Triton's conventions.
+
+This simple online example illustrates that how the OneFlow model can be deployed through Triton and how business modules interact with the Triton server to obtain inference results.
+
+If you want to run this example locally, you can also download [demo.zip](https://oneflow-public.oss-cn-beijing.aliyuncs.com/oneflow-documentation/serving/demo.zip), unzip it and run the file `run.sh` in it.
+
+```shell
+bash run.sh
+```
+
+Next we will introduce the detailed process of OneFlow from training to deployment in detail.
+
+## Analysis of OneFlow Process from Training to Deployment
+
+Firstly we can understand the relationship between OneFlow and Triton in general through the following figure.
+
+![](./imgs/triton-oneflow-backend.png)
+
+As can be seen from the above figure, Triton is in the position of connecting the client and OneFlow: it provides HTTP, gRPC and C interfaces, so that developers can flexibly make an inference request and get the result. 
+
+OneFlow and Model Repository provide Triton in its architecture with back-end reasoning capabilities together. OneFlow provides a corresponding interface to export the trained model to the organizational format that is preset by Triton.
+
+In addition, Triton also provides built-in functions such as task scheduling to ensure performance. For details, please refer to [Triton's official documentation](https://github.com/triton-inference-server/server#features).
+
+After understanding these basic concepts, let's analyze the process of OneFlow from model training to deployment in detail:
+
+- Model save
+- Model deployment
+- Start service
+- Client sends request
+
+### Model Save
+
+The model trained in Graph mode can be directly exported to the format required for deployment through `oneflow.save`; if it is trained in Eager mode, it can be exported to the required format after simple conversion. For details, please refer to [Graph and Deployment](../basics/08_nn_graph.md#graph_5).
```

Comment:
```suggestion
The model trained in Graph mode can be directly exported in the required format for deployment through `oneflow.save`; if it is trained in Eager mode, after simple conversion, it can be exported in the required format. For details, refer to [Graph and Deployment](../basics/08_nn_graph.md#graph_5).
```